mscore/metaedit.cpp

```diff
@@ -19,94 +19,294 @@
+            version->setText(mscoreVrs);
+
+      int scoreVrs = m_score->mscVersion();
+      if (scoreVrs)
+            level->setText(QString::number(scoreVrs, 10)); // in decimal
 
-      level->setValue(score->mscVersion());
-      version->setText(score->mscoreVersion());
-      int rev = score->mscoreRevision();
+      int rev = m_score->mscoreRevision();
       if (rev > 99999)  // MuseScore 1.3 is decimal 5702, 2.0 and later uses a 7-digit hex SHA
             revision->setText(QString::number(rev, 16));
       else
             revision->setText(QString::number(rev, 10));
-      filePath->setText(score->importedFilePath());
 
-      int idx = 0;
+      QString sfilePath(m_score->importedFilePath());
+      if (!sfilePath.isEmpty())
+            filePath->setText(m_score->importedFilePath());
+
+      QString sheet = "QLineEdit { background: transparent; }";
+
       QMapIterator<QString, QString> i(s->metaTags());
-      QGridLayout* grid = static_cast<QGridLayout*>(scrollWidget->layout());
       while (i.hasNext()) {
             i.next();
-            QLabel* label = new QLabel;
-            label->setText(i.key());
-            QLineEdit* text = new QLineEdit(i.value(), 0);
-            connect(text, SIGNAL(textChanged(const QString&)), SLOT(setDirty()));
-            grid->addWidget(label, idx, 0);
-            grid->addWidget(text, idx, 1);
-            ++idx;
+
+            QString s = i.key();
+            if (isSystemTag(s)) {
+                  //QLabel* tag = new QLabel;
+                  QLineEdit *tag = new QLineEdit;
+                  tag->setStyleSheet(sheet); // Couldn't find any other way to remove the background.
+                  tag->setFrame(false);
+                  tag->setFocusPolicy(Qt::NoFocus);
+                  tag->setReadOnly(true);
+                  tag->setToolTip(tr("This is a system tag. It's name can't be modified."));
+
+                  if      (s ==  "platform")       tag->setText(tr("Platform"));
+                  else if (s ==  "movementNumber") tag->setText(tr("Movement number"));
+                  else if (s ==  "movementTitle")  tag->setText(tr("Movement title"));
+                  else if (s ==  "workNumber")     tag->setText(tr("Work number"));
+                  else if (s ==  "workTitle")      tag->setText(tr("Work title"));
+                  else if (s ==  "arranger")       tag->setText(tr("Arranger"));
+                  else if (s ==  "composer")       tag->setText(tr("Composer"));
+                  else if (s ==  "lyricist")       tag->setText(tr("Lyricist"));
+                  else if (s ==  "poet")           tag->setText(tr("Poet"));
+                  else if (s ==  "translator")     tag->setText(tr("Translator"));
+                  else if (s ==  "source")         tag->setText(tr("Source"));
+                  else if (s ==  "copyright")      tag->setText(tr("Copyright"));
+                  else if (s ==  "creationDate")   tag->setText(tr("Creation date"));
```

Comment:
Can't we reuse the strings from score.cpp?

Reply:
This would not work because of the first line of the function  Score::metaTag(const QString& s):
`if (_metaTags.contains(s))
             return _metaTags.value(s);`
It would set the titles to their fields.

Reply:
Huh?

Reply:
Forget that last comment, I just understood what you meant.
Leaving the srings as they were in score.cpp means they would need to be translated over there.
Then, musicxml export wouldn't work with translations.
And using the strings from score.cpp would also leave creationDate as is, instead of changing it to the more user friendly Creation date. (creationDate is just an example.)